microcloud/cmd/microcloud/main_init.go

```diff
@@ -327,33 +327,16 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 // waitForJoin issues a token and instructs a system to request a join,
 // and then waits for the request to either complete or time out.
 // If the request was successful, it additionally waits until the cluster appears in the database.
-func waitForJoin(sh *service.Handler, clusterSize int, secret string, peer string, cfg types.ServicesPut) error {
-	mut := sync.Mutex{}
-	err := sh.RunConcurrent(false, false, func(s service.Service) error {
-		token, err := s.IssueToken(context.Background(), peer)
-		if err != nil {
-			return fmt.Errorf("Failed to issue %s token for peer %q: %w", s.Type(), peer, err)
-		}
-
-		mut.Lock()
-		cfg.Tokens = append(cfg.Tokens, types.ServiceToken{Service: s.Type(), JoinToken: token})
-		mut.Unlock()
-
-		return nil
-	})
-	if err != nil {
-		return err
-	}
-
+func waitForJoin(sh *service.Handler, clusterSize map[types.ServiceType]int, secret string, peer string, cfg types.ServicesPut) error {
```

Comment:
Three functions have been changed in this commit, could you please add a commit for each change with a comment describing the reason for the change. E.g. This function no longer issues a token, why was it moved to `AddPeers`?

Reply:
I've split it up now and added some more info in the commit message. 

Basically I thought that since the token issuing process was becoming more complex, and `AddPeers` was going to handle other components of the existing-cluster detection, it made more sense to move all the related logic over there and keep `waitForJoin` to strictly request a join and wait for it to complete.